Add new workspaces attributes (#833)
hashicorp/terraform-website: content/source/docs/enterprise/api/workspaces.html.md

`data.type`                                   | string  |           | Must be `"workspaces"`.
`data.attributes.name`                        | string  |           | The name of the workspace, which can only include letters, numbers, `-`, and `_`. This will be used as an identifier and must be unique in the organization.
`data.attributes.auto-apply`                  | boolean | `false`   | Whether to automatically apply changes when a Terraform plan is successful, [with some exceptions](../workspaces/settings.html#auto-apply-and-manual-apply).
`data.attributes.description`                 | string  | (nothing) | A description for the workspace.
`data.attributes.file-triggers-enabled`       | boolean | `true`    | Whether to filter runs based on the changed files in a VCS push. If enabled, the `working-directory` and `trigger-prefixes` describe a set of paths which must contain changes for a VCS push to trigger a run. If disabled, any push will trigger a run.

> `data.attributes.source-name`                 | string  | (nothing) | A friendly name for the application or client creating this workspace. If set, this will be displayed on the workspace as "Created via [source-name]".

Reviewer: ```suggestion
`data.attributes.source-name`                 | string  | (nothing) | A friendly name for the application or client creating this workspace. If set, this will be displayed on the workspace as "Created via `<SOURCE NAME>`".
```

Standard placeholder styling.

> `data.attributes.source-url`                  | string  | (nothing) | A URL to either a) an external related resource responsible for creating this workspace, or b) documentation or more info about the application or client creating this workspace.

Reviewer: ```suggestion
`data.attributes.source-url`                  | string  | (nothing) | A URL for the application or client creating this workspace. This can be the URL of a related resource in another app, or a link to documentation or other info about the client.
```
